core/src/main/java/hudson/Util.java

```diff
@@ -901,6 +916,23 @@ public static String encode(@NonNull String s) {
      */
     @NonNull
     public static String rawEncode(@NonNull String s) {
+        return encode(s, uriMap);
+    }
+
+    /**
+     * Encode a single path component for use in an HTTP URL.
+     * Escapes all special characters including those outside
+     * of the characters specified in RFC1738.
+     * Encoded characters are space an {@code "!#%$'&*()+,-./=:;< >?@_[\]^ `{|}~}
```

Comment:
```suggestion
     * All characters outside numbers and letters without diacritic are encoded.
```

Otherwise you need to add "control characters" (0-32) and all characters after code 122 (z).